app/models/patient.rb

```diff
@@ -8,4 +8,15 @@ class Patient
 	has_many :pregnancies
 
   validates_presence_of :name, :primary_phone
+
+  def self.search(name_or_phone_str) # TODO optimize
+    name_matches = Patient.where name: name_or_phone_str
+    primary_matches = Patient.where primary_phone: name_or_phone_str
+    secondary_matches = Patient.where secondary_phone: name_or_phone_str
+    results = []
```

Comment:
You're dead on here. If we're not doing anything with results, we can just eschew the results entirely and return the concat of those three arrays.